ci/__main__.py

```diff
@@ -0,0 +1,686 @@
+installed to access required source code from github repositories.
+You can then do::
+
+    python etstool.py install --runtime=... --toolkit=...
+
+to create a test environment from the current codebase and::
+
+    python etstool.py test --runtime=... --toolkit=...
+
+to run tests in that environment.  You can remove the environment with::
+
+    python etstool.py cleanup --runtime=... --toolkit=...
+
+If you need to make frequent changes to the source, it is often convenient
+to install the source in editable mode::
+
+    python etstool.py install --editable --runtime=... --toolkit=...
+
+You can run all three tasks at once with::
+
+    python etstool.py test_clean --runtime=... --toolkit=...
+
+which will create, install, run tests, and then clean-up the environment.  And
+you can run tests in all supported runtimes and toolkits (with cleanup)
+using::
+
+    python etstool.py test_all
+
+Currently supported runtime values are ``3.6``, and currently
+supported toolkits are ``null``, ``pyqt5``, ``pyside2`` and ``wx``.
+Not all combinations of toolkits and runtimes will work, but the tasks will
+fail with a clear error if that is the case.
+
+Tests can still be run via the usual means in other environments if that suits
+a developer's purpose.
+
+Tasks for generating documentation
+==================================
+
+First, install or update the development environment::
+
+    python etstool.py install --runtime=... --toolkit=...
+
+Then, run the command to build documentation::
+
+    python etstool.py docs --runtime=... --toolkit=...
+
+Changing This File
+------------------
+
+To change the packages installed during a test run, change the dependencies
+variable below.  To install a package from github, or one which is not yet
+available via EDM, add it to the `ci-src-requirements.txt` file (these will be
+installed by `pip`).
+
+Other changes to commands should be a straightforward change to the listed
+commands for each task. See the EDM documentation for more information about
+how to run commands within an EDM enviornment.
+
+Build changelog
+---------------
+To create a first-cut changelog from the news fragments, use this command::
+
+    python etstool.py changelog build
+
+This will update the changelog file. You should review and edit it.
+"""
+
+import glob
+import os
+import subprocess
+import sys
+from contextlib import contextmanager
+from shutil import copy as copyfile, rmtree
+from tempfile import mkdtemp
+
+import click
+
+supported_combinations = {
+    '3.6': {'pyside2', 'pyside6', 'pyqt5', 'pyqt6', 'wx', 'null'},
+}
+
+# Default Python version to use in the commands below if none is specified.
+DEFAULT_RUNTIME = '3.6'
+
+# Default toolkit to use if none specified.
+DEFAULT_TOOLKIT = 'pyqt5'
+
+# Required runtime dependencies. Should match install_requires in setup.py
+dependencies = {
+    "ipython",
+    "matplotlib",
+    "pandas",
+    "pillow",
+    "pyqt5",
```

Comment:
I think we should explicitly declare `pyface` as a dependency, since we directly import from it. We're currently pulling pyface implicitly, probably through `traitsui`.